File: cc/025.cc
// Project Euler Problem 025
//
// What is the first term in the Fibonacci sequence to contain 1000
// digits?

#include <cstdio>
#include <cmath>
#include "bigmath.h"

int main(int argc, const char* argv[]) {
  BigUnsigned target(1);
  for (int i=1000; i > 1; i--) {
    target *= 10;
  }
  BigUnsigned a(1);
  BigUnsigned b(1);
  BigUnsigned* smaller = &a;
  BigUnsigned* larger = &b;
  BigUnsigned* t;
  int i = 3;
  while (true) {
    *smaller += *larger;
    t = smaller;
    smaller = larger;
    larger = t;
    if (*larger >= target) {
      break;
    }
    i++;
  }
  printf("%d\n", i);
}

File: go/problems/025.go
// Project Euler Problem 025
//
// What is the first term in the Fibonacci sequence to contain 1000
// digits?

package problems

import (
	"fmt"
	"math/big"
)

func Problem025() string {
	f1, f2 := big.NewInt(1), big.NewInt(1)
	var i int
	for i = 2; len(f2.String()) < 1000; i++ {
		f1.Add(f1, f2)
		f2, f1 = f1, f2
	}
	return fmt.Sprintf("%d", i)
}

func init() {
	Functions[25] = Problem025
}
